crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -85,16 +83,47 @@ pub fn parse_scarb_config(
             "Failed to read Scarb.toml manifest file, not found in current nor parent directories",
         )?;
 
-    let package = &metadata.packages[0].manifest_metadata.tool;
+    let package_tool_sncast_result = get_package_tool_sncast(&metadata);
+    if package_tool_sncast_result.is_err() {
+        return Ok(CastConfig::default());
+    }
+
+    cast_config_from_package_tool_sncast(package_tool_sncast_result.unwrap(), profile)
+}
+
+pub fn get_package_tool_sncast(metadata: &scarb_metadata::Metadata) -> Result<&Value> {
+    let first_package = metadata
+        .packages
+        .get(0)
+        .ok_or_else(|| anyhow!("No package found in metadata"))?;
+
+    let tool = first_package
+        .manifest_metadata
+        .tool
+        .as_ref()
+        .ok_or_else(|| anyhow!("No field [tool] found in package"))?;
+
+    let tool_sncast = tool
+        .get("sncast")
+        .ok_or_else(|| anyhow!("No field [tool.sncast] found in package"))?;
 
-    let rpc_url = get_property(package, profile, "url")?;
-    let network = get_property(package, profile, "network")?;
-    let account = get_property(package, profile, "account")?;
+    Ok(tool_sncast)
+}
+
+pub fn cast_config_from_package_tool_sncast(
+    package_tool_sncast: &Value,
+    profile: &Option<String>,
+) -> Result<CastConfig> {
+    let rpc_url = get_property_from_profile(package_tool_sncast, profile, "url")?;
+    let network = get_property_from_profile(package_tool_sncast, profile, "network")?;
+    let account = get_property_from_profile(package_tool_sncast, profile, "account")?;
+    let accounts_file = get_property_from_profile(package_tool_sncast, profile, "accounts-file")?;
```

Comment:
it would be cool, if we didn't have to do that, but instead iterate over keys from Scarb.toml and try to match them to CastConfig directly